packages/react-native-web/src/modules/ImageLoader/index.js

```diff
@@ -111,37 +119,78 @@ const ImageLoader = {
       if (typeof failure === 'function') {
         failure();
       }
-      ImageLoader.abort(requestId);
+      ImageLoader.release(requestId);
       clearInterval(interval);
     }
   },
-  has(uri: string): boolean {
+  has(uri: ?string): boolean {
     return ImageUriCache.has(uri);
   },
-  load(uri: string, onLoad: Function, onError: Function): number {
+  load(
+    source: ImageSource,
+    onLoad: (ImageResult) => void,
+    onError: Function
+  ): number {
     id += 1;
     const image = new window.Image();
-    image.onerror = onError;
-    image.onload = (e) => {
+
+    const handleLoad = () => {
       // avoid blocking the main thread
-      const onDecode = () => onLoad({ nativeEvent: e });
-      if (typeof image.decode === 'function') {
-        // Safari currently throws exceptions when decoding svgs.
-        // We want to catch that error and allow the load handler
-        // to be forwarded to the onLoad handler in this case
-        image.decode().then(onDecode, onDecode);
-      } else {
-        setTimeout(onDecode, 0);
-      }
+      const onDecode = () => {
+        ImageUriCache.add(source.uri);
+        onLoad({
+          source: {
+            uri: image.src,
+            width: image.naturalWidth,
+            height: image.naturalHeight
+          }
+        });
+      };
+
+      // Safari currently throws exceptions when decoding svgs.
+      // We want to catch that error and allow the load handler
+      // to be forwarded to the onLoad handler in this case
+      image.decode().then(onDecode, onDecode);
     };
-    image.src = uri;
-    requests[`${id}`] = image;
+
+    image.onerror = onError;
+    image.onload = handleLoad;
+    requests[id] = { image, source };
+
+    // When headers are supplied we can't load the image through `image.src`, but we `fetch` it as an AJAX request
```

Comment:
> we can't load the image through `image.src`

Can you please update the comment to say _why_ this can't be done?

Reply:
Can you help with that, I'm not sure how to describe it 

`image` as you can see is an HTML element (`const image = new window.Image();`)
This element has a `src` attribute that accepts a `string` value - the URL to load 
there's no attribute for `headers` 
So when `headers` are supplied as a parameter we need to use a different strategy to load the image using `fetch` and then assign a local blob url as `src` for the image